### ISSUE TITLE
Allow users to login with their associated email

### ISSUE THREAD

**Issue by llfinch** [huridocs/uwazidocs#4925]
**Is your feature request related to a problem? Please describe.**
Over the last several days, I've personally fielded multiple incidences of our own HURIDOCS colleagues not remembering exactly what their username is for an Uwazi instance (e.g. is it capitalized or all lowercase? is there a white space unaccounted for?) and this becomes a blocker for logging in. I also have heard of partners facing this issue. The only solution at the moment, as far as I can see, is for the user to get in touch with an admin outside of Uwazi to request that the admin confirm their username. Using the Forgot Password? option does not send the user their username. I also have heard of partners facing this issue. In the vast majority of these cases, the user knows which email is associated with their Uwazi user account, however. 

**Describe the solution you'd like**
Allow users to login with either their username or their associated email. This would prevent these lockouts.

**Have you considered an alternative?**
I suppose as an alternative we could expand our Forgot Password? functionality to also include Forgot Username? but that seems more complex... We could also change the flow for Forgot Password? to make sure it sends a reminder of the username to their email, but that seems like extra steps and not straightforward because what a user needs in that moment is to find out their username, not reset their password...

**Screenshots**
What the Forgot Password? prompt sends to your email: 
![image](https://user-images.githubusercontent.com/56080527/180440420-ed4abcc8-effb-4c3e-a44e-7438a45d3f49.png)

**Comment by txau** [comment]
@llfinch not being able to login with your e-mail was done that way for security reasons. E-mails are easy to guess, they are published in websites, used for login in other websites, included in random e-mails threads, etc. Usernames on the other hand can be as obfuscated as needed.

Being able to recover the username is a good alternative.

Another possibility is to have some tweaks around username generation. Ie. only allowing lowercase and making sure we trim any whites paces at the beginning and the end of the usernames. These two cases (casing, whites paces) seem to be the source of most confusions.

**Comment by llfinch** [comment]
@txau Good point! Then having a way built into Uwazi to recover your username would be the ideal, I think. Even if we made tweaks around username generation, I can still see the scenario of people misplacing their username and running into this lockout. So in either case a recovery function would be good.

**Comment by RafaPolit** [comment]
After conversations, we have decided that it is OK to allow to also login with email after all.  With account locking after too many attempts, plus 2fa, it is OK to allow to ALSO use email to log in.

**Comment by iJustErikk** [comment]
Anyone mind if I take this on?

I took a look around and I think this will be simple- where do we document our data models/indices?
